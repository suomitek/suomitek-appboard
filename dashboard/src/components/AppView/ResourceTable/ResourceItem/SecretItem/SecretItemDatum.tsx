import * as React from "react";
import { Eye, EyeOff } from "react-feather";

interface ISecretItemDatumProps {
  name: string;
  value: string;
}

interface ISecretItemDatumState {
  hidden: boolean;
}

class SecretItemDatum extends React.PureComponent<ISecretItemDatumProps, ISecretItemDatumState> {
  // Secret datum is hidden by default
  public state: ISecretItemDatumState = {
    hidden: true,
  };

  public render() {
    const { name, value } = this.props;
    const { hidden } = this.state;
    const decodedValue = atob(value);
    return (
      <span className="flex">
        <button className="secretDatum" onClick={this.toggleDisplay}>
          {hidden ? <Eye /> : <EyeOff />}
        </button>
        <span className="flex margin-l-normal">
          <span>{name}:</span>
          {hidden ? (
            <span className="margin-l-small">{decodedValue.length} bytes</span>
          ) : (
            <pre className="SecretContainer">
              <code className="SecretContent">{decodedValue}</code>
            </pre>
          )}
        </span>
      </span>
    );
  }

  private toggleDisplay = () => {
    this.setState({
      hidden: !this.state.hidden,
    });
  };
}

export default SecretItemDatum;
