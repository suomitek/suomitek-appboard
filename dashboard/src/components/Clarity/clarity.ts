// Code from: https://stackblitz.com/edit/react-ts-wrapper-5w8nhf
import { CdsButton as Button } from "@clr/core/button";
import { CdsIcon as Icon, ClarityIcons as ClrIcons } from "@clr/core/icon-shapes";
import {
  angleIcon,
  applicationsIcon,
  backupRestoreIcon,
  bundleIcon,
  checkCircleIcon,
  circleArrowIcon,
  ClarityIcons as Icons,
  clusterIcon,
  copyToClipboardIcon,
  deployIcon,
  exclamationCircleIcon,
  exclamationTriangleIcon,
  eyeHideIcon,
  eyeIcon,
  fileGroupIcon,
  fileIcon,
  heartIcon,
  helpIcon,
  infoCircleIcon,
  plusCircleIcon,
  refreshIcon,
  rewindIcon,
  searchIcon,
  timesCircleIcon,
  timesIcon,
  trashIcon,
  uploadCloudIcon,
} from "@clr/core/icon-shapes";

import "@clr/core/button";
import "@clr/core/icon";
import { createReactComponent } from "./converter/reactWrapper";

type CdsIconType = Icon;
export const CdsIcon = createReactComponent<CdsIconType>("cds-icon");
export const ClarityIcons = ClrIcons;

type CdsButtonType = Button & HTMLButtonElement;
export const CdsButton = createReactComponent<CdsButtonType>("cds-button");

Icons.addIcons(
  angleIcon,
  bundleIcon,
  checkCircleIcon,
  circleArrowIcon,
  clusterIcon,
  exclamationTriangleIcon,
  eyeHideIcon,
  eyeIcon,
  fileGroupIcon,
  helpIcon,
  infoCircleIcon,
  rewindIcon,
  searchIcon,
  trashIcon,
  uploadCloudIcon,
  applicationsIcon,
  heartIcon,
  exclamationCircleIcon,
  timesIcon,
  deployIcon,
  fileIcon,
  copyToClipboardIcon,
  backupRestoreIcon,
  refreshIcon,
  plusCircleIcon,
  timesCircleIcon,
);
