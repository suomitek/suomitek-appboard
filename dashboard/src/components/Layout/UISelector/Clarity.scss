.color-icon-info {
  cds-icon {
    --color: var(--clr-alert-info-icon-color, #0072a3);
  }
}

.color-icon-success {
  cds-icon {
    --color: var(--clr-alert-success-icon-color, #3c8500);
  }
}

.color-icon-warning {
  cds-icon {
    --color: var(--clr-alert-warning-icon-color, #666666);
  }
}

.color-icon-danger {
  cds-icon {
    --color: var(--clr-alert-danger-icon-color, #c21d00);
  }
}

.center {
  display: grid;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;
}
