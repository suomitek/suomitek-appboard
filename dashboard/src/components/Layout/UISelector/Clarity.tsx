import * as React from "react";

import "../../../clr-ui.min.css";
import "./Clarity.css";

const CSSSelector: React.FC = () => null;

export default CSSSelector;
