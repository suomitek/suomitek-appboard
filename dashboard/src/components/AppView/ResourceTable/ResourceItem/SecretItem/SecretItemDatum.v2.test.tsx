import { mount } from "enzyme";
import * as React from "react";

import { CdsIcon } from "components/Clarity/clarity";
import SecretItemDatum from "./SecretItemDatum.v2";

const testProps = {
  name: "foo",
  value: "YmFy", // foo
};

it("renders the secret datum (hidden by default)", () => {
  const wrapper = mount(<SecretItemDatum {...testProps} />);
  expect(wrapper.find(CdsIcon).findWhere(i => i.prop("shape") === "eye")).toExist();
  expect(wrapper.find(CdsIcon).findWhere(i => i.prop("shape") === "copy-to-clipboard")).toExist();
  expect(wrapper).toMatchSnapshot();
});

it("displays the secret datum value when clicking on the icon", () => {
  const wrapper = mount(<SecretItemDatum {...testProps} />);
  expect(wrapper.find("input").props()).toMatchObject({
    type: "password",
    value: "bar",
  });
  const icon = wrapper.find("button").findWhere(b => b.prop("aria-label") === "Show Secret");
  icon.simulate("click");
  wrapper.update();
  expect(wrapper.find(CdsIcon).findWhere(i => i.prop("shape") === "eye-hide")).toExist();
  expect(wrapper.find("input").props()).toMatchObject({
    type: "text",
    value: "bar",
  });
});
