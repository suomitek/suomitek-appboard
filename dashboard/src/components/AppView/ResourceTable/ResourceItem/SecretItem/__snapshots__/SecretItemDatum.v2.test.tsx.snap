// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the secret datum (hidden by default) 1`] = `
<SecretItemDatum
  name="foo"
  value="YmFy"
>
  <Row
    list={false}
  >
    <div
      className="clr-row"
    >
      <Column
        listItem={false}
        span={5}
      >
        <div
          className="clr-col-5"
        >
          <label
            className="secret-datum-text"
            htmlFor="secret-datum-content-foo"
          >
            foo
          </label>
        </div>
      </Column>
      <Column
        listItem={false}
        span={5}
      >
        <div
          className="clr-col-5"
        >
          <input
            className="clr-input secret-datum-content"
            id="secret-datum-content-foo"
            readOnly={true}
            type="password"
            value="bar"
          />
        </div>
      </Column>
      <Column
        listItem={false}
        span={1}
      >
        <div
          className="clr-col-1"
        >
          <button
            aria-controls="secret-item-datum-foo-ref"
            aria-expanded={false}
            aria-label="Show Secret"
            className="secret-datum-icon"
            onClick={[Function]}
          >
            <ReactWrapperComponent
              shape="eye"
              size="md"
              solid={true}
            >
              <cds-icon />
            </ReactWrapperComponent>
          </button>
        </div>
      </Column>
      <Column
        listItem={false}
        span={1}
      >
        <div
          className="clr-col-1"
        >
          <button
            aria-expanded={false}
            className="secret-datum-icon"
            onClick={[Function]}
          >
            <div
              data-for="app-status"
              data-tip={true}
            >
              <CopyToClipboard
                text="bar"
              >
                <ReactWrapperComponent
                  aria-label="Copy foo secret value to the clipboard"
                  onClick={[Function]}
                  shape="copy-to-clipboard"
                  size="md"
                  solid={true}
                >
                  <cds-icon />
                </ReactWrapperComponent>
              </CopyToClipboard>
            </div>
          </button>
          <div
            style={
              Object {
                "opacity": "0",
              }
            }
          >
            <ReactTooltip
              clickable={false}
              effect="solid"
              id="app-status"
              insecure={true}
              resizeHide={true}
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
                id="app-status"
              >
                Copied
              </div>
            </ReactTooltip>
          </div>
        </div>
      </Column>
    </div>
  </Row>
</SecretItemDatum>
`;
